gnucash/gnome-utils/gnc-tree-util-split-reg.c

```diff
@@ -550,42 +550,9 @@ gnc_tree_util_split_reg_rotate (GncTreeViewSplitReg *view, GtkTreeViewColumn *co
 
     if (viewcol == COL_TYPE)
     {
-        const char type_flags[] = {TXN_TYPE_INVOICE, TXN_TYPE_PAYMENT, 0}; // list of type flags
-        const gchar *flags;
-        const gchar *text;
-        gchar *this_flag;
-        gint index = 0;
-        char type;
-
-        flags = type_flags;
-
-        text = g_strdup_printf("%c", xaccTransGetTxnType (trans));
-
-        /* Find the existing text in the list of flags */
-        this_flag = strstr (flags, text);
-
-        if (this_flag != NULL && *this_flag != '\0')
-        {
-            /* In the list, choose the next item in the list
-               (wrapping around as necessary). */
-            index = this_flag - flags;
-
-            if (flags[index + 1] != '\0')
-                index = index + 1;
-            else
-                index = 0;
-
-            type = type_flags[index];
-        }
-        else
-            type = TXN_TYPE_NONE;
-
-        gnc_tree_view_split_reg_set_dirty_trans (view, trans);
-        if (!xaccTransIsOpen (trans))
-            xaccTransBeginEdit (trans);
-
-        xaccTransSetTxnType (trans, type);
-        return TRUE;
+        // thanks to commit 995f5d3ff, the APAR transaction type is no
+        // longer user modifiable.
+        PERR ("rotating a COL_TYPE viewcol. this shouldn't happen.");
```

Comment:
Let's not have self-referring comments. Neither the comment nor the commit message explain why the transaction type shouldn't be user-modifiable. 

The error message isn't very meaningful either.
```
/* Changing from invoice to payment in the register can create a error condition
  * that can't be resolved from the user interface.
  * /
PERR ("Changing from invoice to payment in the register isn't permitted.");
```

Note that this file is for register2 only and doesn't affect the one that everyone actually uses.

But you do realize that this file is part of Register2, right? And that this function doesn't necessarily apply only to AP/AR accounts?

Reply:
Ok will remove; @mtalexander has removed Register2 in master 3161df6 anyway.

Reply:
> Let's not have self-referring comments. Neither the comment nor the commit message explain why the transaction type shouldn't be user-modifiable.
> 
> The error message isn't very meaningful either.

fwiw the txn_type became unmodifiable due to @Bob-IT nice work in 995f5d3ff because changing TXN_TYPE_NONE to TXN_TYPE_PAYMENT was previously possible (although illegal according to @derekatkins) and would irretrievably damage the book.